src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -212,93 +218,84 @@ public void registerAbilities(List<IFluidTank> abilityList) {
-                    .widget(new ToggleButtonWidget(7, 64, 18, 18,
-                            GuiTextures.BUTTON_LOCK, this::isLocked, this::setLocked)
-                                    .setTooltipText("gregtech.gui.fluid_lock.tooltip")
-                                    .shouldUseBaseBackground());
-        } else {
-            tankWidget = new TankWidget(fluidTank, 69, 52, 18, 18)
-                    .setAlwaysShowFull(true).setDrawHoveringText(false);
-
-            builder.image(7, 16, 81, 55, GuiTextures.DISPLAY)
-                    .widget(new ImageWidget(91, 36, 14, 15, GuiTextures.TANK_ICON))
-                    .widget(new SlotWidget(exportItems, 0, 90, 53, true, false)
-                            .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.OUT_SLOT_OVERLAY));
-        }
-
-        // Add general widgets
-        return builder.label(6, 6, title)
-                .label(11, 20, "gregtech.gui.fluid_amount", 0xFFFFFF)
-                .widget(new AdvancedTextWidget(11, 30, getFluidAmountText(tankWidget), 0xFFFFFF))
-                .widget(new AdvancedTextWidget(11, 40, getFluidNameText(tankWidget), 0xFFFFFF))
-                .widget(tankWidget)
-                .widget(new FluidContainerSlotWidget(importItems, 0, 90, 16, false)
-                        .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.IN_SLOT_OVERLAY))
-                .bindPlayerInventory(entityPlayer.inventory);
-    }
-
-    protected Consumer<List<ITextComponent>> getFluidNameText(TankWidget tankWidget) {
-        return (list) -> {
-            TextComponentTranslation translation = tankWidget.getFluidTextComponent();
-            // If there is no fluid in the tank, but there is a locked fluid
-            if (translation == null) {
-                translation = GTUtility.getFluidTranslation(this.lockedFluid);
-            }
-
-            if (translation != null) {
-                list.add(translation);
-            }
-        };
-    }
-
-    protected Consumer<List<ITextComponent>> getFluidAmountText(TankWidget tankWidget) {
-        return (list) -> {
-            String fluidAmount = "";
-
-            // Nothing in the tank
-            if (tankWidget.getFormattedFluidAmount().equals("0")) {
-                // Display Zero to show information about the locked fluid
-                if (this.lockedFluid != null) {
-                    fluidAmount = "0";
-                }
-            } else {
-                fluidAmount = tankWidget.getFormattedFluidAmount();
-            }
-            if (!fluidAmount.isEmpty()) {
-                list.add(new TextComponentString(fluidAmount));
-            }
-        };
+    @Override
+    public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager) {
+        var fluidSyncHandler = GTFluidSlot.sync(fluidTank)
+                .showAmount(false)
+                .accessibility(true, false)
```

Comment:
Minor, could this be set to allow drain and fill for import hatches, while still being drain only for export hatches?